src/pages/settings/tabs/ApiKeys.tsx

```diff
@@ -0,0 +1,148 @@
+      <div className="flex w-[65%] flex-col gap-y-1">
+        <span className="font-light">{Username}</span>
+        <span className="font-normal">{Device}</span>
+      </div>
+      <Button buttonType="danger" className="px-4 py-2" onClick={onDeleteClick}>
+        Delete API Key
+      </Button>
+    </div>
+  );
+};
+
+const ApiKeys = () => {
+  const [keyValue, setKeyValue] = useState('');
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [, copy] = useCopyToClipboard();
+  const onKeyChange = useEventCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setKeyValue(event.target.value);
+  });
+
+  const {
+    data: createdToken,
+    isError: generatedFailed,
+    isSuccess: generatedSucceed,
+    mutate: createApiToken,
+  } = useCreateApiToken();
+
+  const handleCopyToClipboard = useEventCallback(() => {
+    if (!generatedSucceed) return;
+    copy(keyValue).then((isCopied) => {
+      if (!isCopied) return;
+      toast.success('Copied', 'API Key has been copied to clipboard!', {
+        autoClose: 3000,
+        toastId: 'api-copied',
+      });
+    }).catch(console.error);
+  });
+
+  const onGenerateClick = useEventCallback(() => {
+    createApiToken(keyValue, {
+      onSuccess: () => {
+        setInputDisabled(_ => true);
+        toast.success('API Generated', 'API Key has been generated!', {
+          autoClose: 3000,
+          toastId: 'api-generated',
+        });
+        toast.warning('Copy Your API Key!', 'You won\'t be able to copy this key anymore once you leave this page!', {
+          autoClose: 99999999999,
+          toastId: 'copy-api-key',
+        });
+        invalidateQueries(['auth', 'apikey']);
+      },
+    });
+  });
+
+  const { data: tokens, refetch } = useApiKeyQuery();
+
+  useEffect(() => {
+    if (!generatedSucceed || !createdToken) return undefined;
+    setKeyValue(createdToken);
+    setInputDisabled(true);
+    return () => {
+      if (!toast.isActive('copy-api-key')) return;
+      toast.dismiss('copy-api-key');
+      if (!toast.isActive('api-generated')) return;
+      toast.dismiss('api-generated');
+      if (!toast.isActive('api-copied')) return;
+      toast.dismiss('api-copied');
+    };
+  }, [generatedSucceed, generatedFailed, refetch, createdToken]);
+
+  return (
+    <div className="flex flex-col gap-y-4">
+      <div className="mb-4 text-xl font-semibold">API Keys</div>
+      <div className="text-lg font-semibold">
+        Generate API Key
+      </div>
+      <div className="mb-4 flex flex-row justify-between gap-x-2 border-b-2 border-panel-border pb-8">
+        <Input
+          id="key-input"
+          endIcons={inputDisabled
+            ? [{
+              icon: mdiClipboardTextOutline,
+              className: 'text-panel-text-secondary',
+              onClick: handleCopyToClipboard,
+            }]
+            : undefined}
+          className={cx(!generatedSucceed ? 'w-full xl:w-[65%]' : 'w-[100%]')}
```

Comment:
You need to put `grow` on it not manipulate the field width and put `shrink-0` on a button, maybe adjust flex-basis of a button if needed

Reply:
I fixed this in my branch as I've made some other changes to settings. No need to fix @duehoa1211.